chore: Unsafety policy (#1427)
ariel-os/ariel-os: book/src/coding-conventions.md

### `unsafe` Code

> Code containing `unsafe` is denied outside of modules where the [`unsafe_code`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unsafe-code) lint is explicitly expected

Author: ```suggestion
Code containing `unsafe` is denied outside of modules where the [`unsafe-code`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unsafe-code) lint is explicitly expected
```

> Code containing `unsafe` is denied outside of modules where the [`unsafe_code`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unsafe-code) lint is explicitly expected

Reviewer: ```suggestion
Code containing `unsafe` is denied outside of modules where the [`unsafe_code`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unsafe-code) lint is explicitly `#[expect]`ed
```

(or, in complex situations, allowed).

For all `unsafe` blocks, a `SAFETY` comment MUST be added, in the style [of the `undocumented-unsafe-blocks` Clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/undocumented_unsafe_blocks).

> Any `unsafe` function MUST be documented with the preconditions for sound use in a `# Safety` section, in the style [style of the `missing-safety-doc` Clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#missing_safety_doc).

Reviewer: ```suggestion
Any `unsafe` function MUST be documented with the preconditions for sound use in a `# Safety` section, in the [style of the `missing-safety-doc` Clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#missing_safety_doc).
```
